File: server/src/main/com/broll/gainea/server/core/cards/impl/direct/C_HealingWave.java
package com.broll.gainea.server.core.cards.impl.direct;

import com.broll.gainea.server.core.cards.DirectlyPlayedCard;
import com.broll.gainea.server.core.objects.Unit;
import com.broll.gainea.server.core.utils.UnitControl;

public class C_HealingWave extends DirectlyPlayedCard {

    public C_HealingWave() {
        super(64, "Kristallenergie", "Heilt alle neutralen Einheiten");
    }

    @Override
    protected void play() {
        game.getObjects().stream().filter(it -> it instanceof Unit).map(it -> (Unit) it).filter(Unit::isHurt).forEach(unit -> {
            UnitControl.heal(game, unit, 10000);
        });
    }

}

File: server/src/main/com/broll/gainea/server/core/cards/impl/direct/C_HealingWave.kt
package com.broll.gainea.server.core.cards.impl.direct

import com.broll.gainea.server.core.cards.DirectlyPlayedCard
import com.broll.gainea.server.core.cards.EffectType.OTHER
import com.broll.gainea.server.core.objects.Unit
import com.broll.gainea.server.core.utils.UnitControl.heal

class C_HealingWave :
    DirectlyPlayedCard(64, OTHER, "Kristallenergie", "Heilt alle neutralen Einheiten") {
    override fun play() {
        game.objects.filterIsInstance<Unit>().filter { it.hurt }.forEach { game.heal(it, 10000) }
    }
}
